tests/unit/amo/components/TestAddonBadges.js

```diff
@@ -107,33 +118,34 @@ describe(__filename, () => {
     expect(root.find(Badge)).toHaveLength(0);
   });
 
-  it('displays a badge when the addon is not a web extension', () => {
-    const addon = createInternalAddon(createFakeAddon({
-      files: [{ is_webextension: false }],
-    }));
-    const root = shallowRender({ addon });
+  describe('Quantum compatible badge', () => {
+    let isQuantumCompatible;
 
-    expect(root.find(Badge)).toHaveProp('type', 'not-compatible');
-    expect(root.find(Badge))
-      .toHaveProp('label', 'Not compatible with Firefox Quantum');
-  });
+    beforeEach(() => {
+      isQuantumCompatible = sinon.stub(compatUtils, 'isQuantumCompatible');
+    });
 
-  it('does not display a badge when the addon is a web extension', () => {
-    const addon = createInternalAddon(createFakeAddon({
-      files: [{ is_webextension: true }],
-    }));
-    const root = shallowRender({ addon });
+    it('does not display a badge when add-on is compatible with Quantum', () => {
+      const addon = createInternalAddon(createFakeAddon());
 
-    expect(root.find(Badge)).toHaveLength(0);
-  });
+      isQuantumCompatible.returns(true);
 
-  it('does not display a badge when the addon is not an extension', () => {
-    const addon = createInternalAddon(createFakeAddon({
-      type: ADDON_TYPE_THEME,
-      files: [{ is_webextension: false }],
-    }));
-    const root = shallowRender({ addon });
+      const root = shallowRender({ addon });
 
-    expect(root.find(Badge)).toHaveLength(0);
+      expect(root.find(Badge)).toHaveLength(0);
+    });
+
+    it('displays a badge when the addon is not compatible with Quantum', () => {
+      const addon = createInternalAddon(createFakeAddon());
```

Comment:
I don't think mocking out `isQuantumCompatible` is such a good idea for this case. Since we have no Flow coverage of the tests it's just too hard to keep the mock in sync with the real object. For starters, your tests do not make any assertions to guarantee that `isQuantumCompatible()` is called with the correct arguments. 

I think you should just set up the `addon` with realistic application data instead. This will naturally guarantee that `isQuantumCompatible()` gets called with the right arguments.

Reply:
That's what I did at the beginning but then I duplicated the same `addons` in the compatibility test file..